### PR TITLE
Add admin features

### PR DESCRIPTION
**Pull request by DuncanRose**

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class MoviesController < ApplicationController
   before_action :find_movie, only: [:edit, :update, :show]
+  # before_action :authorize_user, except: [:index, :show]
```
Comment (cmkoller) [inline app/controllers/movies_controller.rb]: Delete commented out line

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class ReviewsController < ApplicationController
+  # before_action :authorize_user, except: [:index, :show]
```
Comment (cmkoller) [inline app/controllers/reviews_controller.rb]: Delete commented out line

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    else
+      flash[:errors] = @user.errors.full_messages
+
+      render
+    end
+  end
+
+  def admin?
+    role == "admin"
+  end
```
Comment (cmkoller) [inline app/controllers/users_controller.rb]: Should this method really be in the UsersController class? Or the User class?

File: spec/support/sign_in_helper.rb [hunk 4]
```diff
@@ -9,4 +9,8 @@ def sign_in_as(user)
 
     click_button 'Log in'
   end
+
+  def create_admin(user)
+    FactoryGirl.create(user, admin: true)
+  end
```
Comment (cmkoller) [inline spec/support/sign_in_helper.rb]: This isn't something you should define a helper method for! Either just write out what you have on line 14 every time you need an admin, or create a specific admin factory ([example here](http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Defining_factories)).

File: spec/features/admins/add_new_admin_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  # I want to add more of me
+  # So that I can sleep
+} do
+  scenario 'give user admin powers' do
+    user = create_admin(:user)
+    sign_in_as user
+
+    visit users_path
+
+    click_button 'Make Admin'
```
Comment (cmkoller) [inline spec/features/admins/add_new_admin_spec.rb]: I don't think this is testing what you want - there's only one user right now and they're already an admin, so you're clicking the "Make Admin" button for yourself?

File: spec/features/admins/add_new_admin_spec.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+feature 'Create new Administrator', %{
+  # As and Admin
+  # I want to add more of me
+  # So that I can sleep
+} do
+  scenario 'give user admin powers' do
+    user = create_admin(:user)
```
Comment (cmkoller) [inline spec/features/admins/add_new_admin_spec.rb]: For clarity, I would suggest naming this variable `admin` to distinguish them from a normal user.

File: spec/features/admins/admin_deletes_movie_spec.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+feature 'admin view users', %{
+  As an admin
+  I want to edit movies
+  So that I can delete them for being bad
+} do
+
+  scenario 'delete individual movies' do
+    movie = FactoryGirl.create(:movie)
+    user = create_admin(:user)
```
Comment (cmkoller) [inline spec/features/admins/admin_deletes_movie_spec.rb]: Naming this user variable `admin` would provide more clarity

File: spec/features/admins/admin_deletes_review_spec.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  scenario 'delete individual reviews' do
+    movie = FactoryGirl.create(:movie)
+    review = FactoryGirl.create(:review, movie: movie, user: user)
+    # binding.pry
+    click_link 'Sign Out'
+
+    user = create_admin(:user)
+    sign_in_as user
```
Comment (cmkoller) [inline spec/features/admins/admin_deletes_review_spec.rb]: Why are you signing in the user if you are just going to sign them out again immediately? Also, the user and the admin should have separate variable names. Thirdly, why does the review need to be attached to a specific user? I would get rid of the `user` let statement all together.

File: spec/features/admins/admin_deletes_users_spec.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  As an admin
+  I want to view a list of users
+  So that I can delete them for being bad
+} do
+
+  scenario 'delete individual user' do
+    user = create_admin(:user)
+    sign_in_as user
+
+    visit users_path
```
Comment (cmkoller) [inline spec/features/admins/admin_deletes_users_spec.rb]: When you visit the `users_path`, won't there be only one user - the admin who's currently signed in? So they're deleting themselves? That's kind of confusing... and actually seems like something that shouldn't be allowed to happen. I think you should try to add a piece that keeps an admin from deleting themselves, and change this test file to both test for that and also test for an admin deleting a different user.

File: spec/features/admins/admin_deletes_review_spec.rb (hunk 8 above)
Comment (DuncanRose) [inline spec/features/admins/admin_deletes_review_spec.rb]: We need the review to be created by other than the admin. hence user creates review and signs out and admin signs in and deletes it, thus checking admins ability to do so..

Comment (cmkoller) [inline spec/features/admins/admin_deletes_review_spec.rb]: If you just say `review = FactoryGirl.create(:review)`, it'll automatically be created by a brand-new user generated by the User factory. There's no need to sign in a user in order for this to happen.

**Pull request merged**